### DISCUSSION TITLE
Periodically getting weird output covering my entire terminal

### DISCUSSION THREAD

**Discussion by hampusolsen** [LazyVim/LazyVim#3909] [Q&A] [answered]
Hello everyone! I've been enjoying LazyVim for a while and it's been great.

_But_, after not doing anything other than using LazyVim's `Sync` for a few days, I have started getting these log outputs which just paints over my entire terminal.

![image](https://github.com/LazyVim/LazyVim/assets/55430265/6f556cf7-c8ca-4f38-9d40-949ad91a8c8a)

Any idea of how I could go about debugging this to find the root cause? :)

**Comment by folke** [comment]
How do you use Neovim?
Lazy thinks you're running headless

**Reply by folke** [reply]
This check:

```lua
function M.headless()
  return #vim.api.nvim_list_uis() == 0
end
```

**Reply by hampusolsen** [reply]
Not sure exactly what you mean by _How do you use Neovim?_, so I'll try to include as much detail as I can!

- I use tmux sessions for different work context, so I will have several Neovim instances running in the different sessions.

- LazyVim is installed as described in the [docs](https://www.lazyvim.org/installation). The only configuration I have done is adding `catppuccin`, disable `inlay_hints` for TypeScript as well as disable the plugin `headlines.nvim`. Oh, and setting `vim.opt.swapfile` to `false`!

- I have a bunch of Lazy Extras installed:
```
coding.mini-surround
coding.yanky
editor.dial
editor.inc-rename
editor.telescope
formatting.prettier
linting.eslint
test.core
util.dot
util.mini-hipatterns

lang.angular
lang.astro
lang.json
lang.markdown
lang.php
lang.toml
lang.typescript
```

**Reply by hampusolsen** [reply]
> This check:
> 
> ```lua
> function M.headless()
>   return #vim.api.nvim_list_uis() == 0
> end
> ```

Will check!

**Reply by hampusolsen** [reply]
You'll have to excuse me, I'm pretty terrible at all this. The check returns `false` for me, and this is how I got the result:

- I added the code below to `~/.config/nvim/lua/config/lazy.lua` and restarted Neovim.

```
function Headless()
  return #vim.api.nvim_list_uis() == 0
end
```

- I called `Headless()` by doing `:lua print(Headless())` and that gave me the result `false`.

![image](https://github.com/LazyVim/LazyVim/assets/55430265/aafe75cb-1112-4134-a109-555a935509bf)

**Reply by folke** [reply]
What I meant is that the output you are seeing will only print when `Config.headless()` returns `true`.

When using Neovim in tui, that check should never return `false`, so it's really weird that it seems to return `false` for you sometimes even when you're clearly in Tui.

**Reply by hampusolsen** [reply, marked as answer]
Oh dear... I think I found the issue and let me start by saying sorry for wasting your time! 🙏🏼 

I had placed a Neovim instance in the background using CTRL-Z, and am now guessing that effectively makes it headless, correct? 😅

**Reply by folke** [reply]
ah yes indeed! :)

**Reply by folke** [reply]
Glad you figured it out

**Reply by hampusolsen** [reply]
Thanks a lot for your time though! Really enjoying the LazyVim experience so far and am looking forward to learning more about Neovim in general~

Cheers 🍻

**Reply by idelice** [reply]
I have the exact same issue. Im running neovim in kitty terminal with no fancy config and this issue have been annoying me lately. 

What's the fix again?

**Reply by dpetka2001** [reply]
Update to latest `lazy.nvim`. I believe [this](https://github.com/folke/lazy.nvim/commit/6fdd904ee45b66d933c5d2f72bcec337e13744f8) should have fixed it. Hopefully, I understand correctly what you mean.

**Reply by hampusolsen** [reply]
> Update to latest `lazy.nvim`. I believe [this](https://github.com/folke/lazy.nvim/commit/6fdd904ee45b66d933c5d2f72bcec337e13744f8) should have fixed it. Hopefully, I understand correctly what you mean.

Unsure if this fixes the issue for instances put in background running previous versions of LazyVim... 🤔

Either way it could be nice for @idelice to either terminate or look at the instance of Neovim to avoid potentially unsaved buffers and what not!

So, @idelice, here is how you could solve the problem without updating (assuming you did the same mistake as I did!):

You most likely have placed a Neovim instance in the background using CTRL+Z, either accidentally or like me... intentionally but forgot about it. 😅

- Check running background jobs:

```sh
$ jobs
[1]  + suspended  nvim
```

- If your output looks like above you can just write `fg` in the terminal to bring it to the foreground. If the list is longer you can specify which job using `fg %job_id`. In the case above that would be `fg %1`.

Let me know if it helped or if you worked it out in another way.

**Reply by folke** [reply]
This was fixed a couple of days ago. no need to do anything

**Comment by idelice** [comment]
Still reproducible as of today

```
[lazy.nvim]         log | 93c9a3f chore(build): auto-generate docs (19 hours ago)
[lazy.nvim]         log | 72c0dc9 fix(git): local plugin fixes (#1624) (19 hours ago)
[lazy.nvim]         log | c771cf4 chore(build): auto-generate docs (27 hours ago)
[lazy.nvim]         log | 25026d2 chore(build): auto-generate docs (27 hours ago)
[lazy.nvim]         log | 083f3df chore(build): auto-generate docs (27 hours ago)
[lazy.nvim]         log | 89b264a chore(main): release 11.10.2 (#1621) (27 hours ago)
[lazy.nvim]         log | 23aeb22 chore(build): auto-generate docs (27 hours ago)
[lazy.nvim]         log | 81d2bff fix(git): only check for new commits for local plugins. Closes #1512 (27 hours ago)
[lazy.nvim]         log | 933f0b5 chore(build): auto-generate docs (2 days ago)
[lazy.nvim]         log | 55b46b3 ci: update (2 days ago)
[lazy.nvim]         log | 23ea80b ci: update (2 days ago)
[lazy.nvim]         log | e6035dc ci: update (2 days ago)
[lazy.nvim]         log | 0ff7e83 chore(build): auto-generate docs (2 days ago)
[lazy.nvim]         log | 1e7745a chore(build): auto-generate docs (2 days ago)
[lazy.nvim]         log | 49a35d3 chore(update): update repository (#1618) (2 days ago)
[lazy.nvim]         log | 16ccd54 chore(build): auto-generate docs (2 days ago)
[lazy.nvim]         log | c060de1 chore(update): update repository (#1616) (2 days ago)
[lazy.nvim]         log | e3154ff chore(build): auto-generate docs (2 days ago)
[lazy.nvim]         log | c3a9cec ci: update (2 days ago)
[lazy.nvim]         log | d901d21 chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | 94b6b67 chore(build): auto-generate docs (3 days ago)[lazy.nvim]         log | 
[lazy.nvim]         log | 538f060 ci: update (3 days ago)
[lazy.nvim]         log | 894cd19 chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | 11e802d chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | 6ca23c1 chore(main): release 11.10.1 (#1612) (3 days ago)
[lazy.nvim]         log | 40e08f2 chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | d0c00e6 ci: remove tests dep (3 days ago)
[lazy.nvim]         log | 61c7156 chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | 6186b3d ci: add generated files to .styluaignore (3 days ago)
[lazy.nvim]         log | 53661bb ci: update (3 days ago)
[lazy.nvim]         log | a1d23e8 chore(build): auto-generate docs (3 days ago)
[lazy.nvim]         log | baac551 fix(lockfile): keep cond=false and enabed=false in lockfile. Fixes #1535. Fixes #1606 (3 days ago)
[lazy.nvim]         log | 407e65c chore(main): release 11.10.0 (#1609) (4 days ago)
[lazy.nvim]         log | 851b120 ci: use main for bootstrap (4 days ago)
[lazy.nvim]         log | 1225f1d ci: dont enable local specs for minit (4 days ago)
[lazy.nvim]         log | a17ad27 build: better minit (4 days ago)
[lazy.nvim]         log | 923e1aa fix(plugin): local spec name (4 days ago)
[lazy.nvim]         log | 6fdd904 fix(config): determine headless only during startup. Fixes #1608 (4 days ago)
[lazy.nvim]         log | 0f2786b feat(profiling): merge VeryLazy stats and show startuptime in profile view (4 days ago)[LazyVim]           log | bf9887a chore(main): release 12.27.0 (#3949) (14 hours ago)
[LazyVim]           log | 28805d1 fix(lazygit): improve git browse (#3941) (14 hours ago)
[LazyVim]           log | 8b2eacb feat(java): allow overriding test config (#3891) (15 hours ago)
[LazyVim]           log | f3c9370 style(icons): CODEOWNERS has a default icon (15 hours ago)
[LazyVim]           log | aa418a2 feat(icons): provide language specific icons in extras (#3931) (16 hours ago)
[LazyVim]           log | 0983141 test: added tests for mini.icons (18 hours ago)
[LazyVim]           log | 33e1da5 fix(fzf-lua): move register_select to lazy init (18 hours ago)
[LazyVim]           log | 6202dd1 chore(main): release 12.26.2 (#3946) (26 hours ago)
[LazyVim]           log | 304729b chore(build): auto-generate docs (27 hours ago)
[LazyVim]           log | d108169 fix(lualine): check that trouble is installed (27 hours ago)
[LazyVim]           log | 4035768 ci: update (2 days ago)
[LazyVim]           log | 21470b4 ci: update (2 days ago)
[LazyVim]           log | 5e12168 ci: update (2 days ago)
[LazyVim]           log | 427b641 chore(update): update repository (#3940) (2 days ago)
[LazyVim]           log | d8958d7 chore(update): update repository (#3938) (2 days ago)
[LazyVim]           log | e985744 chore(main): release 12.26.1 (#3930) (2 days ago)
[LazyVim]           log | ab2ff2e ci: update (2 days ago)
[LazyVim]           log | 3d41040 chore(build): auto-generate docs (2 days ago)
[LazyVim]           log | 5f952bb fix(mini.starter): lazyvim startuptime in mini.starter (#3935) (2 days ago)
[LazyVim]           log | 330d2e4 fix(lualine): use the new ministarter file type to disable in mini.starter (#3929) (3 days ago)
[LazyVim]           log | a915a5d ci: fix urls (3 days ago)
[LazyVim]           log | bb4d4cb ci: update (3 days ago)
[LazyVim]           log | 0471ca1 chore(main): release 12.26.0 (#3920) (3 days ago)
[LazyVim]           log | 4b0f720 feat(vue): enabled hybrid mode to avoid 2 typescript lsp running at same time (#3908) (3 days ago)
[LazyVim]           log | 502dac1 fix(outline): use the correct symbols and filter config format (#3924) (3 days ago)
[LazyVim]           log | 4192d95 fix(dial): move date related augends to default group (#3927) (3 days ago)
[LazyVim]           log | d715456 chore(build): auto-generate docs (3 days ago)
[LazyVim]           log | 6eb8cac ci: update (3 days ago)
[LazyVim]           log | f64bbd2 feat(dial): yaml support (3 days ago)
[LazyVim]           log | 76b41cd feat(util): rest-client (#3915) (3 days ago)
[LazyVim]           log | 51e99df fix(project): properly close window after changing directory (#3919) (3 days ago)
[LazyVim]           log | fb59a9f chore(build): auto-generate vimdoc (3 days ago)
[LazyVim]           log | 8f74db3 refactor(trouble): move options in keymaps for lsp and symbols to `opts` (#3917) (3 days ago)
[LazyVim]           log | d35a391 chore(main): release 12.25.0 (#3905) (4 days ago)[LazyVim]           log | 
[LazyVim]           log | 9c7d3dc feat(erlang): add Erlang language (#3911) (4 days ago)
[LazyVim]           log | 045faec feat(ui): use `mini.icons` instead of `nvim-wev-devicons` (#3899) (4 days ago)
[LazyVim]           log | 1bf8050 ci: new minit (4 days ago)
[LazyVim]           log | c848cfd chore(build): auto-generate vimdoc (4 days ago)
[LazyVim]           log | 9acadc7 fix(fzf): make sure `vim.ui.select` works when fzf is not loaded yet (4 days ago)
[LazyVim]           log | d5f3d39 feat(fzf-lua): preview keymaps for git (#3904) (5 days ago)
```

**Reply by hampusolsen** [reply]
Did you try https://github.com/LazyVim/LazyVim/discussions/3909#discussioncomment-9966682?

**Comment by folke** [comment]
@idelice when do you get this?
I made a change today that should no longer trigger headless mode when you suspended Neovim.
If you however run headless (or with `-l`), then that output is exactly what you should get.
You can configure headless mode, see lazy.nvim settings.

**Reply by idelice** [reply]
`:Lazy` didn't show me any new updates to `lazy` earlier, but now it does. 

I've updated it.

This issue _only_ happens to me when my terminal is not-in-view and i put it back into view.

I'll report back if it happens again.

**Reply by samuelhoover** [reply]
I'm currently facing a similar issue with LazyVim v12.17.0 through the kitty terminal on macOS Sonoma 14.5. I do not have any background jobs running, but https://github.com/LazyVim/LazyVim/discussions/3909#discussioncomment-9957384 does return `false`. It seems to mainly happens whenever I'm away from my computer for a bit as @idelice mentioned.
<img width="1512" alt="image" src="https://github.com/LazyVim/LazyVim/assets/56448170/69e180ce-3bc7-447c-a968-96aa8f551510">

FYI I am running LazyVim through tmux.

**Reply by idelice** [reply]
> I'm currently facing a similar issue with LazyVim v12.17.0 through the kitty terminal on macOS Sonoma 14.5. I do not have any background jobs running, but https://github.com/LazyVim/LazyVim/discussions/3909#discussioncomment-9957384 does return `false`. It seems to mainly happens whenever I'm away from my computer for a bit as @idelice mentioned.
> 
> <img width="1512" alt="image" src="https://github.com/LazyVim/LazyVim/assets/56448170/69e180ce-3bc7-447c-a968-96aa8f551510">
> 
> 
> 
> FYI I am running LazyVim through tmux.

I haven't seen the issue since upgrading lazy to latest, but my guts tells me that your issue is related to running in tmux.

I could be wrong but that's the only difference between our issues, I guess

**Reply by folke** [reply]
@samuelhoover are you 100% certain your lazy.nvim version is the latest main?

**Reply by samuelhoover** [reply]
@folke I think so? I don't see a new update on my end and the version matches the latest release on the lazy.nvim repo. <img width="508" alt="Screenshot 2024-07-10 at 7 05 44 PM" src="https://github.com/LazyVim/LazyVim/assets/56448170/bfb3d737-6c51-489e-8540-87a1a983d70a">

I also think it has something to do with tmux since, but I don't know enough to be sure.
